### PR TITLE
fix(cli): set receipts pruning for chain without deposit contract

### PR DESCRIPTION
**Pull request by emhane**
- Sets receipts prune mode to `PruneMode::Full`, for chains that don't have deposit contract in `ChainSpec`

### CODE REVIEW

File: crates/node/core/src/args/pruning.rs [hunk 1]
```diff
@@ -29,7 +30,8 @@ impl PruningArgs {
                 receipts: chain_spec
                     .deposit_contract
                     .as_ref()
-                    .map(|contract| PruneMode::Before(contract.block)),
+                    .map(|contract| PruneMode::Before(contract.block))
+                    .or(Some(PruneMode::Full)),
```
Comment (mattsse) [inline crates/node/core/src/args/pruning.rs:34]: can we add a note here?

File: crates/node/builder/src/launch/common.rs [hunk 2]
```diff
@@ -339,6 +339,10 @@ impl<R> LaunchContextWith<Attached<WithConfigs, R>> {
 
     /// Returns the configured [`PruneConfig`]
     pub fn prune_config(&self) -> Option<PruneConfig> {
+        let config = self.node_config().prune_config();
+        if config.is_some() {
+            return config
+        }
```
Comment (mattsse) [inline crates/node/builder/src/launch/common.rs:345]: is this an unrelated fix attempt that ranks cli settings higher than what's in the toml?

I don't think we need to do this because we update the toml settings

https://github.com/paradigmxyz/reth/blob/d395df29fcffbc742cd3e819772cc32271ec2d21/crates/node/builder/src/launch/common.rs#L123-L128

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/node/builder/src/launch/common.rs (hunk 2 above)
Comment (emhane) [reply crates/node/builder/src/launch/common.rs:345]: not unrelated, blocks syncing op mainnet full node. see docs for `PruningArgs` flags.

Comment (mattsse) [reply crates/node/builder/src/launch/common.rs:345]: this will ignore all toml settings which I think is incorrect because this always falls back to the default args if run with --full, no?

Comment (emhane) [reply crates/node/builder/src/launch/common.rs:345]: yeah but that's the usual behaviour for any linux tool: cli args have precedence over files. and we have shipped this help for cli `--full` flag since at least January

https://github.com/paradigmxyz/reth/blob/115a58b327b675336c3c033c1443771e5fc6f10e/crates/node/core/src/args/pruning.rs#L12-L15

Comment (mattsse) [inline crates/node/builder/src/launch/common.rs:345]: please undo then this is g2m

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
